canary_tests.py

```diff
@@ -98,7 +98,9 @@ async def test_interrupts_and_pwm(self, pwm):
         should_stop = asyncio.Event()
         ticks = []
         tick_stream = await self.board.stream_ticks([interrupt])
-        counter_task = asyncio.create_task(self.record_tick_data(tick_stream, ticks, should_stop))
+
+        # if we don't return from record_tick_data in five seconds then we fail the test
```

Comment:
Can you add a bit more detailed comment about that if we aren't getting any ticks in the tick_stream the task will never reach the code to exit so we need the timeout?

Reply:
:exploding_head: Oh snap! You're totally right: that would cause it to hang, and there are no bugs in the SDK. Very good catch, you two!